core/src/consensus/tendermint/stake/action_data.rs

```diff
@@ -126,12 +117,106 @@ impl Stakeholders {
     }
 }
 
+pub struct Delegation<'a> {
+    pub delegator: &'a Address,
+    delegatees: BTreeMap<Address, StakeQuantity>,
+}
+
+impl<'a> Delegation<'a> {
+    pub fn load_from_state(state: &TopLevelState, delegator: &'a Address) -> StateResult<Delegation<'a>> {
+        let key = get_delegation_key(delegator);
+        let action_data = state.action_data(&key)?;
+        let delegatees = decode_map(action_data.as_ref());
+
+        Ok(Delegation {
+            delegator,
+            delegatees,
+        })
+    }
+
+    pub fn save_to_state(&self, state: &mut TopLevelState) -> StateResult<()> {
+        let key = get_delegation_key(self.delegator);
+        let encoded = encode_map(&self.delegatees);
+        state.update_action_data(&key, encoded)?;
+        Ok(())
+    }
+
+    pub fn add_quantity(&mut self, delegatee: Address, quantity: StakeQuantity) -> StateResult<()> {
+        *self.delegatees.entry(delegatee).or_insert(0) += quantity;
+        Ok(())
+    }
+
+    #[allow(unused)]
```

Comment:
`unused` is a group of `unused-imports, unused-variables, unused-assignments, dead-code, unused-mut, unreachable-code, unreachable-patterns, unused-must-use, unused-unsafe, path-statements, unused-attributes, unused-macros, unused-allocation, unused-doc-comments, unused-extern-crates, unused-features, unused-labels, unused-parens`.
Use `dead_code` if you just want to allow an unused function.

Or how about `#[cfg(test)]` because it's used in tests.